### PR TITLE
Task Cgroup Model Changes

### PR DESCRIPTION
**Pull request by vsiddharth**
### Summary

Update api model changes

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes:  Yes

### Licensing

This contribution is under the terms of the Apache 2.0 License:  Yes

### CODE REVIEW

File: agent/api/task.go [hunk 1]
```diff
@@ -72,11 +72,11 @@ type Task struct {
 	Containers []*Container
 	// Volumes are the volumes for the task
 	Volumes []TaskVolume `json:"volumes"`
-	// VCPULimit is a task-level limit for compute resources. A value of 1 means that
```
Comment (petderek) [inline agent/api/task.go]: I'm okay with leaving VCPULimit and MemoryLimit on our internal model

File: agent/api/task.go [hunk 2]
```diff
@@ -72,11 +72,11 @@ type Task struct {
...
+	// Cpu is a task-level limit for compute resources. A value of 1 means that
 	// the task may access 100% of 1 vCPU on the instance
-	VCPULimit float64 `json:"CPULimit,omitempty"`
-	// MemoryLimit is a task-level limit for memory resources in bytes
-	MemoryLimit int64 `json:"Memory,omitempty"`
+	Cpu float64 `json:"CPULimit,omitempty"`
+	// Memory is a task-level limit for memory resources in bytes
+	Memory int64 `json:"Memory,omitempty"`
```
Comment (petderek) [inline agent/api/task.go]: We aren't plumbing the ACS model data yet?

Comment (vsiddharth) [reply agent/api/task.go]: The plumbing should be reflection based unless I'm mistaken

File: agent/api/task.go [hunk 3]
```diff
@@ -72,11 +72,11 @@ type Task struct {
 	Containers []*Container
 	// Volumes are the volumes for the task
 	Volumes []TaskVolume `json:"volumes"`
-	// VCPULimit is a task-level limit for compute resources. A value of 1 means that
+	// Cpu is a task-level limit for compute resources. A value of 1 means that
 	// the task may access 100% of 1 vCPU on the instance
-	VCPULimit float64 `json:"CPULimit,omitempty"`
-	// MemoryLimit is a task-level limit for memory resources in bytes
-	MemoryLimit int64 `json:"Memory,omitempty"`
+	Cpu float64 `json:"CPULimit,omitempty"`
```
Comment (richardpen) [inline agent/api/task.go]: I think `CPU` should also work. Can you test these fields in this [file](https://github.com/aws/amazon-ecs-agent/blob/master/agent/acs/handler/acs_handler_test.go#L53)?

File: agent/api/task.go (hunk 1 above)
Comment (vsiddharth) [reply agent/api/task.go]: I can revert back to this if you feel strongly about it.

**Comment by vsiddharth** [comment]
@aaithal thanks a lot for the review!
The functional tests have passed on my instance.

**Pull request merged**